src/librustc_mir/shim.rs

```diff
@@ -768,7 +768,7 @@ fn build_call_shim<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
         }));
     }
 
-    let mut blocks = IndexVec::new();
+    let mut blocks = IndexVec::with_capacity(2);
```

Comment:
You could check `Adjustment::RefMut = rcvr_adjustment` and allocate here with a capacity of `4` if true.

Reply:
Good catch; ithe capacity s actually 5 when this condition is true. I updated the commit.